testing_and_setup/compass/ocean/global_ocean/scripts/create_E3SM_coupling_files.py

```diff
@@ -180,97 +189,113 @@ def graph_partition_ocean(config, mesh_name, date_string, ice_shelf_cavities):
         n = np.concatenate([n, 10**power10 * n_multiples12])
 
     for j in range(len(n)):
-        if n[j] >= min_graph_size and n[j] <= max_graph_size:
+        if min_graph_size <= n[j] <= max_graph_size:
             args = ['gpmetis', 'mpas-o.graph.info.' + date_string, str(n[j])]
             run_command(args)
 
     # create link to output directory
     files = glob.glob('mpas-o.graph.info.*')
-    os.chdir('../assembled_files_for_upload/inputdata/ocn/mpas-o/' + mesh_name)
+    os.chdir('../assembled_files_for_upload/inputdata/ocn/mpas-o/{}'.format(
+        mesh_name))
     for file in files:
-        make_link('../../../../../graph_partition_ocean/' + file, './' + file)
+        make_link('../../../../../graph_partition_ocean/{}'.format(file), file)
 # }}}
 
 
-def initial_condition_seaice(config, mesh_name, date_string, ice_shelf_cavities):
-# {{{
+def initial_condition_seaice(config):  # {{{
 
-    # create links
-    make_link('../init.nc', mesh_name + '.nc')
+    mesh_name = config.get('main', 'mesh_name')
+
+    init_filename = config.get('main', 'initial_condition')
+    base_path = os.path.dirname(os.getcwd())
+    # make it an absolute path with respect to the parent directory if it isn't
+    # already
+    init_filename = os.path.normpath(os.path.join(base_path, init_filename))
+    make_link(init_filename, mesh_name + '.nc')
 
+    filename = 'seaice.{}.nc'.format(mesh_name)
     # command line execution
-    args = ['ncks', '-x', '-v',
-            'bottomDepth,refBottomDepth,restingThickness, temperature,salinity,\
-             temperatureSurfaceValue,salinitySurfaceValue,surfaceVelocityZonal,\
-             surfaceVelocityMeridional,SSHGradientZonal,SSHGradientMeridional,\
-             vertNonLocalFluxTemp,normalVelocity,layerThickness,normalBarotropicVelocity,\
-             vertCoordMovementWeights,boundaryLayerDepth,seaIcePressure,\
-             atmosphericPressure,filteredSSHGradientZonal,filteredSSHGradientMeridional',
-            '-O',  # Overwrite existing file
-            mesh_name + '.nc',
-            'seaice.' + mesh_name + '.nc'
-            ]
+    args = ['ncks', '-v',
+            'areaCell,cellsOnCell,edgesOnCell,fCell,indexToCellID,latCell,'
+            'lonCell,meshDensity,nEdgesOnCell,verticesOnCell,xCell,yCell,zCell,'
+            'angleEdge,cellsOnEdge,dcEdge,dvEdge,edgesOnEdge,fEdge,'
+            'indexToEdgeID,latEdge,lonEdge,nEdgesOnCell,nEdgesOnEdge,'
+            'verticesOnEdge,weightsOnEdge,xEdge,yEdge,zEdge,areaTriangle,'
+            'cellsOnVertex,edgesOnVertex,fVertex,indexToVertexID,'
+            'kiteAreasOnVertex,latVertex,lonVertex,xVertex,yVertex,zVertex',
```

Comment:
I think this is the complete list of fields we want to keep for the sea-ice initial condition but this should *definitely* have another pair of eyes on it.